daemon/daemon.go

```diff
@@ -661,7 +664,26 @@ func createPrefixLengthCounter() *counter.PrefixLengthCounter {
 
 // NewDaemon creates and returns a new Daemon with the parameters set in c.
 func NewDaemon(dp datapath.Datapath) (*Daemon, *endpointRestoreState, error) {
+	var (
+		err           error
+		netConf       *cnitypes.NetConf
+		configuredMTU = option.Config.MTU
+	)
+
 	bootstrapStats.daemonInit.Start()
+
+	if option.Config.ReadCNIConfiguration != "" {
```

Comment:
Was this deliberately placed before option Config Validation?

Reply:
No, good catch. Fixed.